src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -152,9 +152,12 @@ private async ValueTask<IChecksummedPersistentStorage> CreatePersistentStorageAs
             // some sort of issue (like DB corruption).  We'll then try to delete the DB and can
             // try to create it again.  If we can't create it the second time, then there's nothing
             // we can do and we have to store things in memory.
-            return await TryCreatePersistentStorageAsync(solutionKey, workingFolderPath).ConfigureAwait(false) ??
-                   await TryCreatePersistentStorageAsync(solutionKey, workingFolderPath).ConfigureAwait(false) ??
-                   NoOpPersistentStorage.Instance;
+            var result = await TryCreatePersistentStorageAsync(solutionKey, workingFolderPath).ConfigureAwait(false) ??
+                         await TryCreatePersistentStorageAsync(solutionKey, workingFolderPath).ConfigureAwait(false);
+            if (result != null)
+                return result;
+
+            return NoOpPersistentStorage.Instance;
```

Comment:
no change in logic.  this just makes it easier to set a breakpoint on the failure case when debugging issues.